src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList/v1/TaskList.tsx

```diff
@@ -20,11 +20,13 @@ interface TaskListProps {
 
 // used two sort two Tasks based on ascending times
 const sortAscendingDate = (task1: Task, task2: Task) => {
-  const deadLine1 = task1.deadline.getTime();
-  const deadLine2 = task2.deadline.getTime();
-
-  if (deadLine1 !== deadLine2) return deadLine1 - deadLine2;
-  return task1.dateCreated.getTime() - task2.dateCreated.getTime();
+  if (task1.deadline && task2.deadline) {
+    const deadLine1 = task1.deadline.getTime();
+    const deadLine2 = task2.deadline.getTime();
+    if (deadLine1 !== deadLine2) return deadLine1 - deadLine2;
+    return task1.dateCreated.getTime() - task2.dateCreated.getTime();
+  }
```

Comment:
I think we should return whichever one that has a deadline ahead of one that doesnt if one has it and the other doesnt